rrpproxy/rrpproxy.py

```diff
@@ -8,7 +8,11 @@
 
 
 class RRPProxy:
-    def __init__(self, username, password, use_test_environment=False):
+    def __init__(self, username, password, use_test_environment=False, session=None):
+        self.session = session
+        if not session:
+            self.session = requests
```

Comment:
Because you are assigning `session` directly to `self.session` it might be more user friendly to just have a `use_session` switch. That saves the user having to do an extra import. For example:
```
class RRPProxy:
    session = requests.Session()
    def __init__(self, username, password, use_test_environment=False, use_session=False):
        self.session = session if use_session else requests
```

Reply:
The RRPProxy only allows 5 connections per registrar, which is why I have chosen to pass along your own session instead of creating one for you. 
This allows for this construction, which automatically closes the connection again:
```python
with Session() as s:
    client = RRPProxy(username, password, session=s)
```
Otherwise you need to make sure you specifically close your session using `client.session.close()` in this case.